tests/rootfs.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/sh
+
+set -eu
+
+ERROR=1
+OK=0
+MY_NAME=$(basename -- "$0")
+ROOTFS_DIRECTORY="./rootfs"
+TEST_IMAGE_FILE_PATH="${ROOTFS_DIRECTORY}/tmp/test.img"
+TEST_EMPTY_MTAB_FILE="${ROOTFS_DIRECTORY}/etc/mtab"
+
+test_setup()
+{
+    if [ ! -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static" ]
+    then
+        cp "$(command -v qemu-arm-static)" "${ROOTFS_DIRECTORY}/usr/bin/"
+    fi
+
+    dd if=/dev/zero of="${TEST_IMAGE_FILE_PATH}" bs=2M count=64 >/dev/null 2>&1 || return ${ERROR}
+    touch "${TEST_EMPTY_MTAB_FILE}" >/dev/null 2>&1 || return ${ERROR}
+    mkfs.ext4 "${TEST_IMAGE_FILE_PATH}" >/dev/null 2>&1 || return ${ERROR}
+    mount -o bind /dev/ "${ROOTFS_DIRECTORY}/dev/" >/dev/null 2>&1 || return ${ERROR}
+}
+
+test_destroy()
+{
+    umount -l "${ROOTFS_DIRECTORY}/dev"
+
+    if [ -f "${TEST_IMAGE_FILE_PATH}" ]
+    then
+        rm -f ${TEST_IMAGE_FILE_PATH}
+    fi
+
+    if [ -f "${TEST_EMPTY_MTAB_FILE}" ]
+    then
+        rm -f "${TEST_EMPTY_MTAB_FILE}"
+    fi
+    if [ -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static" ]
+    then
+        rm -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static"
+    fi
+}
+
+run_test()
+{
+    if ! test_setup
+    then
+        printf "Cannot run tests, unable to complete test setup\\n"
+        test_destroy
+        exit ${ERROR}
+    fi
+
+    if "$1"
```

Comment:
hmm, you run the setup and teardown for each test ... it is not overly bad, but can we not use a cascading scheme?

test_setup
runtest1
runtest2
...
test_teardown

would safe on wear and tear on my flash drive :( I don't want to run this on my flash storage for sure :)
if you do prefer to have a setup each time, put the img in a tmpfs/ramdisk instead! unlimited super fast bring-up and teardown. Actually much more preferable!

Reply:
Yes every test should be run with a clean slate. 
But creating the file in tmp is a good idea